web-common/src/runtime-client/batched-request.ts

```diff
@@ -0,0 +1,96 @@
+import type {
+  QueryServiceQueryBatchBody,
+  V1QueryBatchEntry,
+  V1QueryBatchResponse,
+} from "@rilldata/web-common/runtime-client/gen/index.schemas";
+import { DefaultQueryPriority } from "@rilldata/web-common/runtime-client/http-request-queue/priorities";
+import { streamingFetchWrapper } from "@rilldata/web-common/runtime-client/streaming-fetch-wrapper";
+
+export type BatchRequest = {
+  request: V1QueryBatchEntry;
+  priority: number;
+  resolve: (data: V1QueryBatchResponse) => void;
+  reject: (err: Error) => void;
+};
+export class BatchedRequest {
```

Comment:
I'd expect a Batcher to self-manage when to fire HTTP requests. Like, users of this class should only be calling `batcher.add(myRequest)`, not `batcher.send()`.

Internal to this class, I'd expect to see logic for when a batch is full or when a time window has passed – after which the request would be issued. `send()` would be a private method, not a public one (unless it's a public `forceSend()`).